pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2320,6 +2320,11 @@ func handleDelete(ctx *domainContext, key string, status *types.DomainStatus) {
 	// No point in publishing metrics any more
 	ctx.pubDomainMetric.Unpublish(status.Key())
 
+	err := hyper.Task(status).Delete(status.DomainName)
```

Comment:
On the line above (2300) there is a call `doInactivate()` which calls `Delete()`. Do you have any idea why it was not called so that you need to make introduce another `Delete()` call?

I'm looking into the `Delete()` logic precisely and how it is being called and it seems it is broken in all the code (obviously not your fault, I just want to share some concerns).

I see this repeating call pattern: Delete(), Cleanup(). (callers: verifyStatus() and doActivateTail()). `Cleanup()` waits for qmp to be gracefully deleted by the qemu itself (`waitForQmp()`) and considers this as 'qemu has gone' event, but if `Delete()` deletes the whole "kvm" folder, then obviously `waitForQmp` fails immediately and returns, because it gets an error on attempt to connect to a qmp socket. So this graceful `waitForQmp()` logic never worked as expected.

Reply:
The call is not to KvmContext:Delete.  The previous call to Delete is to ctrdContext:Delete.  I uploaded (probably too many) log files.  Like this entry (I did add in some special logs).

```
{
  "severity": "warning",
  "source": "zedbox",
  "iid": "1515",
  "content": "{
    "file":"/pillar/hypervisor/containerd.go:180",
    "func":"github.com/lf-edge/eve/pkg/pillar/hypervisor.ctrdContext.Delete",
    "level":"warning",
    "msg":"ctrdContext:Delete  10614140-bf96-4f0d-b37b-591e85f74d11.1.2 goroutine:
        github.com/lf-edge/eve/pkg/pillar/hypervisor.ctrdContext.Delete()
            /pillar/hypervisor/containerd.go:180
        github.com/lf-edge/eve/pkg/pillar/hypervisor.ctrdContext.Cleanup()
            /pillar/hypervisor/containerd.go:204
        github.com/lf-edge/eve/pkg/pillar/hypervisor.KvmContext.Cleanup()
            /pillar/hypervisor/kvm.go:986
        github.com/lf-edge/eve/pkg/pillar/cmd/domainmgr.doCleanup()
            /pillar/cmd/domainmgr/domainmgr.go:1786
        github.com/lf-edge/eve/pkg/pillar/cmd/domainmgr.doInactivate()
            /pillar/cmd/domainmgr/domainmgr.go:1782
        github.com/lf-edge/eve/pkg/pillar/cmd/domainmgr.handleDelete()
            /pillar/cmd/domainmgr/domainmgr.go:2300
        github.com/lf-edge/eve/pkg/pillar/cmd/domainmgr.runHandler()
            /pillar/cmd/domainmgr/domainmgr.go:889",
    "pid":1515,
    "source":"zedbox",
    "time":"2024-02-29T23:41:20.962516887Z\"
  }",
  "msgid": 4630,
  "timestamp": {
    "seconds": 1709250080,
    "nanos": 962516887
  },
  "filename": "/pillar/hypervisor/containerd.go:180",
  "function": "github.com/lf-edge/eve/pkg/pillar/hypervisor.ctrdContext.Delete"
}
```
This is the only call to any Delete, made by ctrdContext:Cleanup to its Delete.  Which will not clean up the directory.  Maybe I should move the call to KvmContext:Delete here.

Reply:
I built and tested the `xen` hypervisor.  Seems to working.

Reply:
I checked the logs Bob shared with me. Logs contain the following:
```
msg":"xl destroy failed:  13d7afb3-dc67-4eb0-9031-6720f02ce988.1.2 is an invalid domain identifier (rc=-6)
; couldn't delete task 13d7afb3-dc67-4eb0-9031-6720f02ce988.1.2: CtrLoadContainer: Exception while loading container: container \\"13d7afb3-dc67-4eb0-9031-6720f02ce988.1.2\\" in namespace \\"eve-user-apps\\": not found",
```
this presumably comes from the `Delete()` function for the xen hyper. The function is not idempotent and complains on a second attempt to delete the container if it was deleted. There are other similar errors start with "xl destroy failed" prefix. So has to be fixed.